cdap-app-fabric/src/main/java/io/cdap/cdap/internal/events/EventPublishManager.java

```diff
@@ -41,7 +42,8 @@ public class EventPublishManager extends AbstractIdleService {
   @Inject
   EventPublishManager(CConfiguration cConf, Set<EventPublisher> eventPublishers,
                       EventWriterProvider eventWriterProvider) {
-    this.publishEnabled = cConf.getBoolean(Constants.Event.PUBLISH_ENABLED);
+    this.publishEnabled = new DefaultFeatureFlagsProvider(cConf).isFeatureEnabled(
+      Feature.EVENT_PUBLISH.getFeatureFlagString());
```

Comment:
for feature flag please use the Feature EVENT_PUBLISH in this case and pass the provider to it's isEnabled method

Reply:
Done